### ISSUE TITLE
Rotation?

### ISSUE THREAD

**Issue by Efreak** [zb3/kf#1]
I find the way pieces are rotating around a certain piece hard to get used to. You need to keep track of which piece is the center. Instead, how about rotating around the center? When you rotate twice (ignoring the middle step), the two pieces should switch places instead of having one of them move over two spots.

I can't figure out (from a quick search) if this is the intended Puyo Puyo rotation or not, so perhaps it should also be an option.

Also, the pieces should probably be coming down from the middle of the field regardless of how big it is. Currently if you make the field wider, they still come down in the third-left column.

**Comment by zb3** [comment]
Hi, thanks for your feedback!

Of course, pieces should be coming down from the middle, I forgot to change that after I've made an option to choose arbitrary field size -> now fixed!

About the rotation - I've seen this way everywhere, so I've implemented it this way.
Currently it's something like this:
1
2

**Comment by zb3** [comment]
Hi, thanks for your feedback!

Of course, pieces should be coming down from the middle, I forgot to change that after I've made an option to choose arbitrary field size -> now fixed!

About the rotation - I've seen the current way everywhere, so I've implemented it this way.
I don't know if I understood you correctly, but I think you're talking about this:

```
1
2 

1 2

2
1

2 1
```

Am I right?
I've added this as option as suggested - should work, try it :)

**Comment by Efreak** [comment]
Yes, that's what I'm talking about for rotation. Like tetris. I don't play many games of this type, and it seems illogical to me to be rotating it around anything other than the center of mass.

Unfortunately, whatever you've done has broken the game, at least in Chrome. I get this error:
`Uncaught TypeError: Cannot read property 'length' of null` from [Display.js#77](https://github.com/zb3/kf/blob/366e4c463e415637327df9f7b950b6bf8f138e5a/js/Display.js#L77)

**Comment by zb3** [comment]
I wasn't able to reproduce, but I've found a case where this line would fail - injected cross domain stylesheets, so I've fixed it for that case.

However, this case would have nothing to do with my changes :O 
So could you tell if my fix fixed the issue?

**Comment by Efreak** [comment]
Hmm. Same error. I should note that I'm running Chrome Dev build, not stable/beta (I can't add my own extensions/userscripts in these :-1:). I get the same error when I start a game or try to change the board size. 

The game does work fine under a week-old firefox nightly (I'm downloading update now, will post if it changes after), however, so perhaps the problem is on my end. And yes, the changes I had requested work fine.

It also works fine using IE 11, though I notice that it pauses slightly after matching 4 blocks, before it continues; Firefox does not.

Unfortunately, the first computer I had used this game on was my laptop, and I no longer have that device. I should have stable versions of chrome/firefox installed as soon as I can figure out how to separate their profiles (and in chrome's case the browser itself as well).
